pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,10 +184,87 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_report(self):
+        """Get maintenance report from maintenance repository
+
+        Returns:
+            :class:`~pubtools.pulplib.MaintenceReport` object describes the
+            maintenance status
+
+            None if there's no report in the repository
+        """
+        try:
+            data = self._do_get_maintenance().result()
+        except requests.exceptions.HTTPError:
+            # if the maintenance mode hasn't been used before, then it could return
+            # HTTPError 404.
+            LOG.warning("No maintenance info found")
+            return None
+
+        return MaintenanceReport.from_data(data)
+
+    def set_maintainenance(self, regex, enable=True, **kwargs):
```

Comment:
A question, why do we have this public method for setting maintenance mode given some repo IDs, *and* MaintenanceReport also has public methods for adjusting the report given repo IDs?  Which one do we expect clients to use?

I think the cleaner design would be to have responsibilities split up so that:

- methods on MaintenanceReport are responsible for creating a new & modified report
- this method is only responsible for uploading a MaintenanceReport

It also gives you a nice symmetry - the client has a getter which returns a MaintenanceReport, then it also has a setter which returns a MaintenanceReport too.

Reply:
hm, I guess I got what you mean\